app/views/users/show.html.erb

```diff
@@ -5,5 +5,10 @@
     <ul class="posts">
       <%= render @posts %>
     </ul>
+    <%= @user_friends.each do |u| %>
+      <% if !find_friendship(current_user.id, u.id).nil? %>
+        <%= link_to 'hello', user_path(u.id) %>
+      <% end %>
+    <% end %>
```

Comment:
Kindly move logic away from views. View-related logic can be encapsulated in View helpers (app/helpers/\*_helper.rb) 
Check [this](https://mixandgo.com/learn/the-beginners-guide-to-rails-helpers) guide to get started with helpers.